orange_grove_final_colorize.rb

```diff
@@ -0,0 +1,92 @@
+require "colorize"
+
+# create orange grove class
+class OrangeGrove
+  attr_reader :grove, :total_oranges
+
+  # create empty grove array
+  def initialize
+    @grove = []
+  end
+
+  # user passes number of trees in grove
+  def plant_grove(num_trees)
+    num_trees.times do
+      @grove.push(OrangeTree.new)
+    end
+  end
+
+  # counts all oranges on all trees in array
+  def count_all_the_oranges
```

Comment:
Thought experiment:

I think there is a sequence of events where ```grove.total_oranges``` and ```grove.count_all_the_oranges``` could return different values. 

When might that happen and why?